flytekit/control_plane/nodes.py

```diff
@@ -170,42 +170,42 @@ def task_executions(self) -> List["flytekit.control_plane.tasks.executions.Flyte
         return self._task_executions or []
 
     @property
-    def workflow_executions(self) -> List["flytekit.control_plane.workflow_executions.FlyteWorkflowExecution"]:
-        return self._workflow_executions or []
+    def subworkflow_node_executions(self) -> Dict[str, "flytekit.control_plane.nodes.FlyteNodeExecution"]:
+        return (
+            {}
+            if self._subworkflow_node_executions is None
+            else {n.id.node_id: n for n in self._subworkflow_node_executions}
+        )
 
     @property
-    def executions(self) -> _artifact_mixin.ExecutionArtifact:
-        return self.task_executions or self.workflow_executions or []
+    def executions(self) -> List[_artifact_mixin.ExecutionArtifact]:
+        return self.task_executions or list(self.subworkflow_node_executions.values()) or []
```

Comment:
is the latter half of this just self._subworkflow_node_executions? why the transformations?

Reply:
`_subworkflow_node_executions` is a Dict mapping node id to node execution, so it needs to be transformed to output a list of node executions.

We could alternatively have `executions` output a `Union[List, Dict]` depending on whether it's task executions of subworkflow node executions.

Reply:
wait isn't it a list? https://github.com/flyteorg/flytekit/pull/503/files#diff-655eacdaa79dd7e114e503367646d8831b8a31c032c4e3c295f41f21b71a7863R307

Reply:
but the property `subworkflow_node_executions` is a Dict (see above)... we don't have to convert it into a dict, but I found it convenient to be able to access the subworkflow node executions by node_id

Reply:
right but we don't necessarily need to access these through the property since it's all in the same class right? sorry this is a minor performance nit and not really worth refactoring :)